memdx/packetwriter.go

```diff
@@ -26,54 +26,54 @@ func (pw *PacketWriter) WritePacket(w io.Writer, pak *Packet) error {
 
 	if pak.Magic == MagicReq || pak.Magic == MagicRes {
 		if extFramesLen > 0 {
-			return protocolError{"cannot use framing extras with non-ext packets"}
+			return invalidArgError{"cannot use framing extras with non-ext packets"}
 		}
 
 		if keyLen > math.MaxUint16 {
-			return protocolError{"key too long to encode"}
+			return invalidArgError{"key too long to encode"}
 		}
 
 		binary.BigEndian.PutUint16(headerBuf[2:], uint16(keyLen))
 	} else if pak.Magic == MagicReqExt || pak.Magic == MagicResExt {
 		if extFramesLen > math.MaxUint8 {
-			return protocolError{"framing extras too long to encode"}
+			return invalidArgError{"framing extras too long to encode"}
```

Comment:
But this one, and most of the others here?  The client of Protostellar has no control over the framing extras, it's not exposed in the protocol at all.  If an error is hit here, is it not an internal CNG bug, rather than an InvalidArgs case?

Reply:
That's true but this isn't CNG or protostellar, the callers of this memdx package aren't necessarily even gocbcorex. You'd hope any of these errors could be caught in testing well before a user had chance to see them too.

Reply:
Oh ok.  I think is yet another place where I can't really offer a useful review.  I'll +2 so you're not blocked but please get it reviewed by a competent adult at some point.

Reply:
I think that is a useful review, and it's a place where I don't think that there is strictly a correct answer. We can always selectively change it back...